icu_tokenizer.c uses deprecated ubrk_safeClone

Building yaz-5.34.0 on OS X 10.13 with external dependencies via fink, including libicu-72.1, I see:

```
/bin/sh ../libtool  --tag=CC   --mode=compile gcc -DHAVE_CONFIG_H -I.  -I/sw/include -I/sw/include/libxml2 -I/sw/include -I/sw/include -DYAZ_HAVE_ICU=1 -I./libstemmer_c/include -I/sw/include  -Os -D_THREAD_SAFE  -MT libyaz_icu_la-icu_tokenizer.lo -MD -MP -MF .deps/libyaz_icu_la-icu_tokenizer.Tpo -c -o libyaz_icu_la-icu_tokenizer.lo `test -f 'icu_tokenizer.c' || echo './'`icu_tokenizer.c
libtool: compile:  gcc -DHAVE_CONFIG_H -I. -I/sw/include -I/sw/include/libxml2 -I/sw/include -I/sw/include -DYAZ_HAVE_ICU=1 -I./libstemmer_c/include -I/sw/include -Os -D_THREAD_SAFE -MT libyaz_icu_la-icu_tokenizer.lo -MD -MP -MF .deps/libyaz_icu_la-icu_tokenizer.Tpo -c icu_tokenizer.c  -fno-common -DPIC -o .libs/libyaz_icu_la-icu_tokenizer.o
icu_tokenizer.c:72:21: warning: 'ubrk_safeClone_72' is deprecated [-Wdeprecated-declarations]
    tokenizer->bi = ubrk_safeClone(old->bi, NULL, &bufferSize, &status);
                    ^
/sw/include/unicode/urename.h:511:24: note: expanded from macro 'ubrk_safeClone'
#define ubrk_safeClone U_ICU_ENTRY_POINT_RENAME(ubrk_safeClone)
                       ^
/sw/include/unicode/uvernum.h:123:47: note: expanded from macro 'U_ICU_ENTRY_POINT_RENAME'
#       define U_ICU_ENTRY_POINT_RENAME(x)    U_DEF2_ICU_ENTRY_POINT_RENAME(x,U_ICU_VERSION_SUFFIX)
                                              ^
/sw/include/unicode/uvernum.h:122:51: note: expanded from macro 'U_DEF2_ICU_ENTRY_POINT_RENAME'
#       define U_DEF2_ICU_ENTRY_POINT_RENAME(x,y) U_DEF_ICU_ENTRY_POINT_RENAME(x,y)
                                                  ^
/sw/include/unicode/uvernum.h:121:50: note: expanded from macro 'U_DEF_ICU_ENTRY_POINT_RENAME'
#       define U_DEF_ICU_ENTRY_POINT_RENAME(x,y) x ## y
                                                 ^
<scratch space>:15:1: note: expanded from here
ubrk_safeClone_72
^
/sw/include/unicode/ubrk.h:320:1: note: 'ubrk_safeClone_72' has been explicitly marked deprecated here
U_DEPRECATED UBreakIterator * U_EXPORT2
^
/sw/include/unicode/umachine.h:116:29: note: expanded from macro 'U_DEPRECATED'
#define U_DEPRECATED U_CAPI U_ATTRIBUTE_DEPRECATED
                            ^
/sw/include/unicode/umachine.h:96:52: note: expanded from macro 'U_ATTRIBUTE_DEPRECATED'
#    define U_ATTRIBUTE_DEPRECATED __attribute__ ((deprecated))
                                                   ^
1 warning generated.
```

ICU's docs say this function was deprecated as of version 69.

--- a/src/icu_tokenizer.c
+++ b/src/icu_tokenizer.c
@@ -61,7 +61,6 @@ static void icu_tokenizer_reset(struct icu_tokenizer *tokenizer,
 
 struct icu_tokenizer *icu_tokenizer_clone(struct icu_tokenizer *old)
 {
-    int32_t bufferSize = U_BRK_SAFECLONE_BUFFERSIZE;
     UErrorCode status = U_ZERO_ERROR;
     struct icu_tokenizer * tokenizer
         = (struct icu_tokenizer *) xmalloc(sizeof(struct icu_tokenizer));
@@ -69,7 +68,7 @@ struct icu_tokenizer *icu_tokenizer_clone(struct icu_tokenizer *old)
     assert(old);
     icu_tokenizer_reset(tokenizer, old->action);
     assert(old->bi);
-    tokenizer->bi = ubrk_safeClone(old->bi, NULL, &bufferSize, &status);
+    tokenizer->bi = ubrk_clone(old->bi, &status);
     if (U_SUCCESS(status))
         return tokenizer;
     return tokenizer;

